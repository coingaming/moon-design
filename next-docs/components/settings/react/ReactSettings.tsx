import { useState } from 'react';
import {
  LogoSportsbetShort,
  LogoBitcasinoShort,
  LogoMoonDesignShort,
  LogoBetaddaShort,
} from '@heathmont/moon-assets';
import useFeatureFlags from '../../../utils/useFeatureFlags';
import { useDocsTheme, DocsBrands } from '../../themes/DocsThemeProvider';
import useThemeTW from '../../themes/useThemesTW';
import ReactElixirSwitcher from '../common/ReactElixirSwitcher';
import SettingsButton from '../common/SettingsButton';
import DarkLightModeSwitcher from './DarkLightModeSwitch';
import RTLModeSwitch from './RTLModeSwitch';

type LogosKeys = keyof typeof Logos;

interface Props {
  toggleReactAndElixir: () => void;
  isReactEnabled: boolean;
}

const Logos = {
  sportsbet: <LogoSportsbetShort fontSize="1rem" />,
  bitcasino: <LogoBitcasinoShort fontSize="1rem" />,
  moonDesign: <LogoMoonDesignShort fontSize="1rem" />,
  betadda: <LogoBetaddaShort fontSize="1rem" />,
};

const ReactSettings: React.FC<Props> = ({
  toggleReactAndElixir,
  isReactEnabled,
}) => {
  const [isOpened, setIsOpened] = useState(false);
  const toggle = () => setIsOpened(!isOpened);

  const { toggleColorScheme, setBrand, themeKeys } = useDocsTheme();

  const [isRtlEnabled, setIsRtl] = useState(false);

  const { isFeatureElixirEnabled } = useFeatureFlags();

  const toggleRtl = () => {
    const htmlTag = document && document?.getElementsByTagName('html')[0];
    setIsRtl(!isRtlEnabled);
    const dirAttr = htmlTag.getAttribute('dir');

    if (!dirAttr) {
      htmlTag.setAttribute('dir', 'rtl');
    } else {
      htmlTag.removeAttribute('dir');
    }
  };

  const filteredThemeKeys = themeKeys.filter((themeKey: string) =>
    ['sportsbet', 'bitcasino', 'moonDesign', 'betadda'].includes(themeKey)
  );

  const {
    setBrand: setBrandTW,
    toggleMode: toggleModeTW,
    getMode,
  } = useThemeTW();

  const switchBrandHandler = (themeKey: LogosKeys) => {
    setBrand(themeKey as DocsBrands);
    setBrandTW(themeKey);
  };

  const switchModeHandler = () => {
    toggleColorScheme();
    toggleModeTW();
  };
  return (
    <>
      <SettingsButton onClick={toggle} isRtlEnabled={isRtlEnabled} />
      {isOpened && (
        <>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-16 z-50`}
          >
            {filteredThemeKeys.map((themeKey: string) => {
              const Logo = Logos[themeKey as LogosKeys] || Logos.moonDesign;
              return (
                <button
                  key={themeKey}
                  type="button"
                  onClick={() => switchBrandHandler(themeKey as LogosKeys)}
                  className={`${
                    isRtlEnabled ? `mr-4` : `ml-4`
                  } p-2 rounded-full text-black bg-white hover:bg-gray-200 inline-flex items-center justify-center shadow-md`}
                >
                  {Logo}
                </button>
              );
            })}
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-28 z-50`}
          >
            <DarkLightModeSwitcher
              toggle={switchModeHandler}
              isEnabled={getMode() === 'dark'}
              isRtlEnabled={isRtlEnabled}
            />
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-40 z-50`}
          >
            <RTLModeSwitch toggle={toggleRtl} isEnabled={isRtlEnabled} />
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-52 z-50`}
          >
            {isFeatureElixirEnabled ? (
              <ReactElixirSwitcher
                toggle={toggleReactAndElixir}
                isReactEnabled={isReactEnabled}
              />
            ) : null}
          </div>
        </>
      )}
    </>
  );
};

export default ReactSettings;
