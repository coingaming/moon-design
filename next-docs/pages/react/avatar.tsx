import React from 'react';

import { ThemeProvider, sportsbetDark } from '@heathmont/moon-themes';

export default function PageAvatar() {
  return (
    <>
      <ThemeProvider theme={sportsbetDark}>
        <h1>Avatar</h1>
      </ThemeProvider>
    </>
  );
}
