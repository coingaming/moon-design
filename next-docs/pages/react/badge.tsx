import React from 'react';
import Head from 'next/head';

import { Badge } from '@heathmont/moon-core';
import { ThemeProvider, sportsbetDark } from '@heathmont/moon-themes';

export default function PageBadge() {
  return (
    <>
      <h1>Badge</h1>
      <p>Small count and labeling component.</p>
      <Badge>Active</Badge>
      <Badge color="goten.100" backgroundColor="piccolo.100">
        Active
      </Badge>
    </>
  );
}
