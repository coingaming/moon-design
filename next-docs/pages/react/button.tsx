import React from 'react';
import Head from 'next/head';

import { Button } from '@heathmont/moon-components';
import PreviewSwitch from '../../components/codePreview/Switch';
import Preview from '../../components/codePreview/Preview';

export default function PageBadge() {
  return (
    <>
      <section className="mt-6">
        <h1 className="text-5xl font-semibold">Button</h1>
        <p className="text-lg mt-4">
          Buttons are calls-to-action used to prompt users. They encourage users
          to interact with us in multiple ways throughout our galaxy, based on
          what the label of the button indicates.
        </p>

        <p className="text-lg mt-4">
          Buttons are clickable elements with label text that describe the
          action that will happen when the users interact with it.
        </p>
      </section>
      <section className="mt-6">
        <Preview
          title="Primary"
          preview={<Button variant="primary">Active</Button>}
          code={`import { Button } from '@heathmont/moon-components';

<Button variant="primary">Active</Button>`
}
        />
      </section>
    </>
  );
}
