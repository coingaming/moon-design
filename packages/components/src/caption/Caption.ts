import styled from 'styled-components';
import { rem, themed } from '@heathmont/moon-utils';
import { ColorProps } from '@heathmont/moon-themes';

type Props = {
  color?: ColorProps;
};

const Caption = styled.span<Props>(({ color, theme }) => ({
  color: themed('color', color)(theme),
  fontSize: rem(10),
  lineHeight: rem(16),
  fontWeight: theme.fontWeight.semibold,
  textTransform: 'uppercase',
  letterSpacing: rem(1),
}));

Caption.defaultProps = {
  color: 'bulma.100',
};

export default Caption;
