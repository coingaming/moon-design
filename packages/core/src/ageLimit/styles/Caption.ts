import { rem } from '@heathmont/moon-utils';
import styled from 'styled-components';

const Caption = styled.span(({ theme, color }) => ({
  color,
  fontSize: rem(14),
  lineHeight: 1,
  fontWeight: theme.newTokens.font.weight.semibold,
}));

export default Caption;
