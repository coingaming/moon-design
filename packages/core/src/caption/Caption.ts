import styled from 'styled-components';
import { rem, themed } from '@heathmont/moon-utils';
import { ColorProps } from '@heathmont/moon-themes';

type Props = {
  color?: ColorProps;
  textAlign?: TextAlign;
};

const Caption = styled.span<Props>(
  ({ color, theme }) => ({
    color: themed('color', color)(theme),
    fontSize: rem(10),
    lineHeight: rem(16),
    fontWeight: theme.fontWeight.semibold,
    textTransform: 'uppercase',
    letterSpacing: rem(1),
  }),
  ({ textAlign }) => textAlign && { textAlign },
);

Caption.defaultProps = {
  color: 'bulma.100',
};

export default Caption;
