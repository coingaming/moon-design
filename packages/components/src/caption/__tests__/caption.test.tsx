import * as React from 'react';
import { create } from 'react-test-renderer';
import 'jest-styled-components';
import { sportsbetDark, ThemeProvider } from '@heathmont/moon-themes';

import Caption from '../Caption';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={sportsbetDark}>{component}</ThemeProvider>
);

describe('Caption', () => {
  describe('Elements', () => {
    test('renders as <span> by default', () => {
      const caption = create(renderWithTheme(<Caption>Caption</Caption>));
      expect(caption).toMatchSnapshot();
    });
    test('renders as an element if "as" prop provided', () => {
      const text = create(
        renderWithTheme(<Caption as="p">Caption with Paragraph tag</Caption>)
      );
      expect(text).toMatchSnapshot();
    });
  });

  test('renders default color', () => {
    const caption = create(renderWithTheme(<Caption>Caption</Caption>));
    expect(caption).toMatchSnapshot();
  });

  test('renders with a color', () => {
    const caption = create(
      renderWithTheme(<Caption color="piccolo.100">Caption</Caption>)
    );
    expect(caption).toMatchSnapshot();
  });
});
