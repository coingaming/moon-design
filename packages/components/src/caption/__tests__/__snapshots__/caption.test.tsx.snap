// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption Elements renders as <span> by default 1`] = `
.c0 {
  font-size: 0.625rem;
  line-height: 1rem;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.0625rem;
  -moz-letter-spacing: 0.0625rem;
  -ms-letter-spacing: 0.0625rem;
  letter-spacing: 0.0625rem;
  color: #FFFFFF;
}

<span
  className="c0"
  color="bulma.100"
>
  Caption
</span>
`;

exports[`Caption renders default color 1`] = `
.c0 {
  font-size: 0.625rem;
  line-height: 1rem;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.0625rem;
  -moz-letter-spacing: 0.0625rem;
  -ms-letter-spacing: 0.0625rem;
  letter-spacing: 0.0625rem;
  color: #FFFFFF;
}

<span
  className="c0"
  color="bulma.100"
>
  Caption
</span>
`;

exports[`Caption renders with a color 1`] = `
.c0 {
  font-size: 0.625rem;
  line-height: 1rem;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.0625rem;
  -moz-letter-spacing: 0.0625rem;
  -ms-letter-spacing: 0.0625rem;
  letter-spacing: 0.0625rem;
  color: #0CD463;
}

<span
  className="c0"
  color="piccolo.100"
>
  Caption
</span>
`;
